compiler/damlc/lib/DA/Cli/Damlc.hs

```diff
@@ -861,68 +960,60 @@ execMigrate projectOpts inFile1_ inFile2_ mbDir =
+          [(pkgName1, _pkgId1, lfPkg1, sdkVer1), (pkgName2, _pkgId2, lfPkg2, sdkVer2)] <-
               forM [inFile1, inFile2] $ \inFile -> do
                   bytes <- B.readFile inFile
                   let dar = ZipArchive.toArchive $ BSL.fromStrict bytes
                   -- get the main pkg
                   dalfManifest <- either fail pure $ readDalfManifest dar
                   let pkgName = takeBaseName $ mainDalfPath dalfManifest
-                  let dalfs = dalfPaths dalfManifest
-                  deps <-
-                      forM dalfs $ \dalfPath -> do
-                          dalf <- getEntry dalfPath dar
-                          (pkgId, _pkg) <- decode $ BSL.toStrict $ ZipArchive.fromEntry dalf
-                          pure (pkgId, stringToUnitId $ takeBaseName dalfPath)
                   mainDalfEntry <- getEntry (mainDalfPath dalfManifest) dar
                   (mainPkgId, mainLfPkg) <- decode $ BSL.toStrict $ ZipArchive.fromEntry mainDalfEntry
-                  pure (pkgName, mainPkgId, mainLfPkg, MS.fromList deps)
-          let pkgMap = MS.union depMap1 depMap2
+                  pure (pkgName, mainPkgId, mainLfPkg, sdkVersion dalfManifest)
           -- generate upgrade modules and instances modules
           let eqModNames =
                   (NM.names $ LF.packageModules lfPkg1) `intersect`
                   (NM.names $ LF.packageModules lfPkg2)
+          thisSdkVer <- getSdkVersion
+          let differentSdks = (length $ nubSort [sdkVer1, sdkVer2, thisSdkVer]) /= 1
           let eqModNamesStr = map (T.unpack . LF.moduleNameString) eqModNames
-          let buildOptions =
+          let buildOptions0 =
                   [ "--allow-different-sdks"
-                  , "'--package=" <> show (pkgName1, [(m, m ++ "A") | m <- eqModNamesStr]) <> "'"
+                  , "'--package=" <> show ("instances-" <> pkgName1
+                                          , [(m', m' ++ "A")
+                                            | m <- eqModNamesStr, let m' = m <> "Instances"
+                                            ]
+                                          ) <> "'"
+                  , "'--package=" <> show ("instances-" <> pkgName2
+                                          , [(m', m' ++ "B")
+                                            | m <- eqModNamesStr, let m' = m <> "Instances"
+                                            ]
+                                          ) <> "'"
+                  ]
+          let buildOptions1 =
+                  [ "'--package=" <> show (pkgName1, [(m, m ++ "A") | m <- eqModNamesStr]) <> "'"
                   , "'--package=" <> show (pkgName2, [(m, m ++ "B") | m <- eqModNamesStr]) <> "'"
-                  , "--ghc-option=-Wno-unrecognised-pragmas"
                   ]
+          let buildOptions
+                  | differentSdks = buildOptions0 ++ buildOptions1
+                  | otherwise = buildOptions1
```

Comment:
If we have different sdks we need to allow this with the flag --allow-different-sdks and also import the created instances modules with aliases. If not, we only need aliases for the modules of the two equally named packages. You can't add aliases for the instances modules in the later case, because they are not generated. I'll add a comment.